### ISSUE TITLE
Manage Groups: Show 10 per page does not render pagination

### ISSUE THREAD

**Issue by DiemBTran** [scientist-softserv/palni-palci#275]
# Summary

When managing groups from `Dashboard` > `Manage Groups`, the `Show 10 per page` option does not render pagination. So if there are 11 groups that exist, a user cannot see the 11th group unless they `Show 25 per page`.

# Acceptance Criteria

- [ ] There is pagination on `Dashboard` > `Manage Groups` so that users can click the next page to view groups number 11 and above

# Screenshots or Video

There are 11 groups on this tenant, but you cannot see the 11th group "test" unless you "Show 25 per page":

https://user-images.githubusercontent.com/29311858/200934658-5f5519c7-aeba-441b-90cc-933bd31feb40.mp4

# Testing Instructions
- [ ] as an admin user, navigate to `Dashboard` > `Manage Groups`
- [ ] if there are not already 11 groups, create however many groups you need to reach 11 total
- [ ] verify that once 11 groups exist, pagination renders on this page so that you can view the 11th group

**Comment by DiemBTran** [comment]
Linked items:

Relates to:

- [3]Test Groups and Roles

**Comment by DiemBTran** [comment]
Lea Ann - Sept 2022:

---

This is a bug in dev.commons-archive.org
https://dev.commons-archive.org/admin/groups?locale=en

**Comment by DiemBTran** [comment]
Jeremy - Sept 2022:

---

When we don't have a query string, we rely on `Hyrax::Group.includes(:roles).order("roles.sort_value")`.  We are attempting to sort on a field that is part of the `LEFT JOIN` of a query.  The creates challenges on getting the correct number of records (which is part of pagination and how Rails handles the `includes` imperative).

Let's say we have three groups (A, B, and C).  And we have three roles (1, 2, 3).  And A has roles 1 and 2; B has roles 2; and C has roles 3.  The pseudo result set for count will say we have three groups (e.g. A, B, C).  However, when we do the eager load necessary for the `includes` directive (which provides the `left_outer_join` functionality), we have a result set that looks like the following: `[[A,1], [A,2], [B,2], [B,3]]`. And if our pagination window is 2, then the `left_outer_join` logic will say `"SELECT * FROM groups WHERE ID IN (?, ?)", "a", "a")` (because those are the two records that we get back in our query).

tl;dr sorting on a field that we bring in via a left outer join is challenging given the pagination and includes loading that Rails provides.

**Comment by DiemBTran** [comment]
Jeremy - Sept 2022:

---

You'll want to look at the specific SQL statements that are generated for the two queries of a paginator: the count and the paginated result set.

**Comment by DiemBTran** [comment]
Jeremy - Sept 2022:

---

A possible solution is to cache the sort value on the group; when we add a role, cache the smallest sort_value of all the group's roles on the group record.

Alternatively, we can explore writing the appropriate SQL, however a challenge for that will be that there are two queries for pagination; a count to determine the size of the results and a results set query (that has the offset and limits).  This is doable, but will require an alternate approach that separates the eager loading of `include` and the determination of the sort_order.

(Eg. write a subquery that is part of the order clause or some such antics)

**Comment by DiemBTran** [comment]
Lea Ann - Sept 2022:

---

This ticket will need to be done before contributing the groups/permissions work back to Hyku. This is not a bug introduced with the Hyku Upgrade, so removing that label.

**Comment by DiemBTran** [comment]
Lea Ann - Sept 2022:

---

The file in question is `app/models/hyrax/group.rb`
